### PR TITLE
[ci] support authorized shas for external PRs

### PR DESCRIPTION
**Pull request by cseed**
- add ci database with 1 table: authorized_shas
 - only start build if authorized author or source_sha is authorized
 - form on main page to add authorized sha,
 - unauthorized PRs are created, but not tested, and shown as unauthorized in UI

### CODE REVIEW

File: ci/ci/github.py [hunk 1]
```diff
@@ -143,6 +139,16 @@ def __init__(self, number, title, source_repo, source_sha, target_branch, author
         self.target_branch.batch_changed = True
         self.target_branch.state_changed = True
 
+    async def authorized(self, pool):
```
Comment (danking) [inline ci/ci/github.py]: poll is super vague, can we call this `dbpool`? or maybe just `db`?

File: ci/ci/github.py [hunk 2]
```diff
@@ -362,11 +365,14 @@ def github_status(self):
...
                 self.target_branch.state_changed = True
 
-    async def _heal(self, batch_client, on_deck):
+    async def _heal(self, batch_client, pool, on_deck):
         # can't merge target if we don't know what it is
         if self.target_branch.sha is None:
             return
 
+        if not await self.authorized(pool):
+            return
```
Comment (danking) [inline ci/ci/github.py]: looks good, this should always protect the only call to `_start_build`. We gotta be careful in future changes that we never add other paths to `_start_build`.

File: ci/deployment.yaml [hunk 3]
```diff
@@ -55,15 +55,18 @@ spec:
...
-              name: hail-ci-0-1-github-oauth-token
-            - mountPath: /ci-jwt
-              readOnly: true
-              name: ci-jwt
-            - name: jwt-secret-key
-              mountPath: /jwt-secret-key
-              readOnly: true
+           - name: ci-user-secret
+             mountPath: /ci-user-secret
+             readOnly: true
```
Comment (danking) [inline ci/deployment.yaml]: if you make changes anyway: symmetry: readOnly first, and maybe fix `jwt-secret-key` to be the same way

**Comment by cseed** [comment]
`db` sounds like a connection.  I used dbpool.

Added a check in _start_build. I had debated about this, but I agree, certainly safer.

I put the yaml fields in alphabetical order, which is what the K8s docs usually do.

**Pull request merged**